### PR TITLE
Ff Divide/Conquer

### PR DESCRIPTION
**Pull request by jdgarciauc3m**
FF support for divide and conquer

### CODE REVIEW

File: include/ff/parallel_execution_ff.h [hunk 1]
```diff
@@ -62,18 +63,24 @@ class parallel_execution_ff {
   and ordering.
   \param concurrency_degree Number of threads used for parallel algorithms.
   \param order Whether ordered executions is enabled or disabled.
+  \note The concurrency_degree is a hint and the implementation might select
+        a lower value.
   */
   parallel_execution_ff(int concurrency_degree, bool order = true) noexcept :
-      concurrency_degree_{concurrency_degree},
+      concurrency_degree_{std::min(concurrency_degree, 
+          static_cast<int>(std::thread::hardware_concurrency()))},
```
Comment (mdolz) [inline include/ff/parallel_execution_ff.h]: Does this mean that FF does not support oversubscription? Why?

**Review by javidoc00 — CHANGES_REQUESTED** [review]
Sample "sort_sequence" gives an error of pattern not implemented.

**Review by mdolz — CHANGES_REQUESTED** [review]
Sort sequence example doesn't work with ff back end:

$ ./sort_sequence 10 ff
terminate called after throwing an instance of 'std::runtime_error'
  what():  Pattern divide_conquer not implemented

**Review by Drioastorga — CHANGES_REQUESTED** [review]
Sort sequence sample uses the old version of the D&C interface. Consider updating the sample to use the new interface in order to support ff as backend for this sample.

Fibbonacci sample using the ff backend is extremely slow. Is this normal?

Why is limited the concurrency degree to the hardware concurrency?. This is Ok only if FF does not support a higher number of threads that the number of cores.

File: include/ff/parallel_execution_ff.h (hunk 1 above)
Comment (jdgarciauc3m) [reply include/ff/parallel_execution_ff.h]: When increasing concurrency degree the error obtained is:

ERROR: FARM, try to add too many workers, please increase max_nworkers

Comment (mdolz) [reply include/ff/parallel_execution_ff.h]: Ok.

**Review by mdolz — DISMISSED** [review]
Ok.

**Review by javidoc00 — APPROVED** [review]

**Review by mdolz — APPROVED** [review]
Ok.

**Review by Drioastorga — APPROVED** [review]
Ok

**Pull request merged**